metadata-ingestion/examples/library/dataset_add_domain_execute_graphql.py

```diff
@@ -1,15 +1,31 @@
 # read-modify-write requires access to the DataHubGraph (RestEmitter is not enough)
 from datahub.ingestion.graph.client import DatahubClientConfig, DataHubGraph
+from datahub.emitter.mce_builder import make_dataset_urn, make_domain_urn
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.emitter.rest_emitter import DatahubRestEmitter
+from datahub.metadata.schema_classes import DomainsClass, ChangeTypeClass
+
+# Define the dataset URN
+dataset_urn = make_dataset_urn(platform="hive", name="realestate_db.sales", env="PROD")
 
 gms_endpoint = "http://localhost:8080"
-graph = DataHubGraph(DatahubClientConfig(server=gms_endpoint))
+rest_emitter = DatahubRestEmitter(gms_server=gms_endpoint)
+
+# Define the domain URN
+domain_urn = make_domain_urn("marketing")
+
+# Create the Domains aspect
+domains_aspect = DomainsClass(domains=[domain_urn])
```

Comment:
we'd prefer people use GraphQL, precisely because we future proofed the model here but don't want to publicly expose it in usual scenarios. is there an impetus for adding this?

Reply:
If someone is emitting other aspects using the rest emitter (which we recommend in other docs), I would want them to be able to have a similar pattern for all their writes so they don't need to switch back and forth between patterns.

Reply:
This is actually what I faced. I use emitter for other things and for domains I am using gql which does not match with my entire util functions